include/zephyr/bluetooth/gatt.h

```diff
@@ -1558,6 +1558,7 @@ struct bt_gatt_discover_params;
  *  @param conn Connection object.
  *  @param attr Attribute found, or NULL if not found.
  *  @param params Discovery parameters given.
+ *  @param err 0 on success, -ECONNRESET on connection lost, -EINVAL on invalid data
```

Comment:
A bit out of scope of this PR, but maybe relevant to the documentation here: I think we should disconnect the ATT bearer when the peer violates protocol. In case of the fixed channel UATT bearer, we should disconnect the whole ACL connection, like we do for ATT timeout.

If we do the disconnect, should we still have a separate error here for protocol violation, or should we say it also results in `ECONNRESET` here?